v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -265,3 +269,192 @@ function getTaskFromCliArguments(
 
   return task;
 }
+
+function parseTaskArguments(
+  cliArguments: string[],
+  usedCliArguments: boolean[],
+  task: Task,
+): Record<string, any> {
+  const taskArguments: Record<string, unknown> = {};
+
+  parseNamedParameters(cliArguments, usedCliArguments, task, taskArguments);
+
+  parsePositionalAndVariadicParameters(
+    cliArguments,
+    usedCliArguments,
+    task,
+    taskArguments,
+  );
+
+  usedCliArguments.forEach((arg, index) => {
```

Comment:
`.forEach` just adds overhead without any obvious benefit. A pattern like this would be better:

```js
      const asd = [false, false, false];
      const unsedIndex = asd.indexOf(false);
      if (unsedIndex !== -1) {
        throw new Error('asd');
      }
```

Please don't read this as "this is too slow". My point is about not using something that has a cost for now benefit.

Reply:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5300/commits/eb56b333ad3c3cbcc0f3904d73ef797f89689971)